File: CheckIn.Api.Common/Models/Lists/SubtaskCombinedListModel.cs
using System.ComponentModel.DataAnnotations;
using CheckIn.Api.Common.Models.Interfaces;

namespace CheckIn.Api.Common.Models.Lists;

public record SubtaskCombinedListModel : IEntityModel
{
    public Guid Id { get; init; }
    public bool IsCompleted { get; init; }
    public Guid ResponseGroupId { get; set; }
    [MaxLength(255)] public string? RespondentName { get; init; }
    [MaxLength(255)] public string? Comment { get; init; }
    public Guid? AssignedToUserId { get; init; }

    [MaxLength(255)] public string? AssignedToEmail { get; set; }
    public Guid? CompletedByUserId { get; init; }
    public DateTime? CompletedAt { get; init; }
    public bool IsGeneratedFromTask { get; set; }
    public DateTime CreatedAt { get; init; }
    public DateTime Deadline { get; init; }

    public required string Title { get; init; }
    public string? Description { get; init; }
    public Guid TemplateSubtaskId { get; init; }
}

File: checkin_frontend/lib/features/task_overview/data/models/subtask_combined_list_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'subtask_combined_list_model.freezed.dart';
part 'subtask_combined_list_model.g.dart';

@freezed
sealed class SubtaskCombinedListModel with _$SubtaskCombinedListModel {
  const factory SubtaskCombinedListModel({
    required String id,               // ID Inštancie
    @Default(false) bool isCompleted,
    required String responseGroupId,
    String? respondentName,
    String? comment,
    String? assignedToUserId,
    String? completedByUserId,
    DateTime? completedAt,
    required String title,
    String? assignedToEmail,
    required DateTime deadline,
    DateTime? createdAt,
    String? description,              // Popis zo šablóny
    required String templateSubtaskId,
    @Default(false) bool isGeneratedFromTask,
  }) = _SubtaskCombinedListModel;

  factory SubtaskCombinedListModel.fromJson(Map<String, dynamic> json) =>
      _$SubtaskCombinedListModelFromJson(json);
}